pkg/apis/pipeline/v1beta1/resultref.go

```diff
@@ -26,10 +26,12 @@ import (
 type ResultRef struct {
 	PipelineTask string `json:"pipelineTask"`
 	Result       string `json:"result"`
+	Property     string `json:"property"`
 }
 
 const (
-	resultExpressionFormat = "tasks.<taskName>.results.<resultName>"
+	resultExpressionFormat       = "tasks.<taskName>.results.<resultName>"
+	objectResultExpressionFormat = "tasks.<taskName>.results.<objectResultName>.<individualAttribute>"
```

Comment:
It may be worth adding a comment here about some of the discussion in this section of the TEP https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md#collisions-with-builtin-variable-replacement

Reply:
sgtm! Thanks @lbernick

Reply:
I'm still a bit confused here about what this comment is trying to say; is this accurate?

```
Result expressions of the form <resultName>.<attribute> will be treated as object results. If a string result name contains a dot, brackets should be used to differentiate it from an object result.
```

Also, could you link to the specific section of the TEP? (Much easier for someone reading than to have to look through two full TEPs.)

Lastly, has this support already been added? If not, that should probably be done first, so users have an option to express results with dots in the names before the way we interpret that syntax changes.

Reply:
> I'm still a bit confused here about what this comment is trying to say; is this accurate?
> 
> ```
> Result expressions of the form <resultName>.<attribute> will be treated as object results. If a string result name contains a dot, brackets should be used to differentiate it from an object result.
> ```
> 
> Also, could you link to the specific section of the TEP? (Much easier for someone reading than to have to look through two full TEPs.)

Sounds great to me!! Thanks for the suggestion.

Reply:
> Lastly, has this support already been added? If not, that should probably be done first, so users have an option to express results with dots in the names before the way we interpret that syntax changes.

Do you mean if `If a string result name contains a dot, brackets should be used to differentiate it from an object result.` is already supported?

Reply:
Another question related to this. Do we really want to support the result name that contains dots(.)? The [result regex ](https://github.com/tektoncd/pipeline/blob/33b3eac346a37ae7943f210fa13d9c4ea7d571fa/pkg/apis/pipeline/v1beta1/resultref.go#L40) seems supporting that, but [parseExpression](https://github.com/tektoncd/pipeline/blob/33b3eac346a37ae7943f210fa13d9c4ea7d571fa/pkg/apis/pipeline/v1beta1/resultref.go#L125-L131) in the validation webhook seems not supporting this, which is inconsistent.

Also https://github.com/tektoncd/pipeline/issues/3590 seems only to proposed supporting param names that contain dots.

If we don't want dots in result names, we can change the [result regex ](https://github.com/tektoncd/pipeline/blob/33b3eac346a37ae7943f210fa13d9c4ea7d571fa/pkg/apis/pipeline/v1beta1/resultref.go#L40) to make sure result name cannot contain dots. This seems the quick solution for the conflicts with `objectResultName.key` name.

Reply:
having dot in result name is problematic for parsing.

Reply:
I will handle this case later

Reply:
@Yongxuanzhang thanks!

@lbernick do you think it's okay to merge this pr first? Thanks!

Reply:
update: talked to @lbernick @Yongxuanzhang @jerop about this offline. [TEP-0080](https://github.com/tektoncd/community/blob/main/teps/0080-support-domainscoped-parameterresult-names.md) proposed to support using dots in both param and result names. Param part was supported but not result. 

#4910 and #4909 are the tickets tracking result names containing dots.

Reply:
I am a little surprised since this PR https://github.com/tektoncd/pipeline/pull/4268 fixed the results. 🤔

Reply:
Thanks @pritidesai for the pointer.
Seems like #4268 has implemented the variable replacement part on the reconciler side for result names that contain dots and are referenced via bracket notation. But only the parsing in the validation webhook doesnt recognize `$(tasks.myTask.results["foo.bar"]` (https://github.com/tektoncd/pipeline/issues/4909#issuecomment-1138080594)

Reply:
After chatting with @chuangw6 I realized this is actually not a breaking change-- it allows more types of params to be validated than before, and doesn't change existing validation

Reply:
After chatting with @chuangw6 I realized this is actually not a breaking change-- it allows more types of params to be validated than before, and doesn't change existing validation